web/src/Web.App/Extensions/WebApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Web.App.Infrastructure.Session;
+
+namespace Web.App.Extensions;
+
+public static class WebApplicationBuilderExtensions
+{
+    public static WebApplicationBuilder AddCacheService(this WebApplicationBuilder builder)
+    {
+        var section = builder.Configuration.GetSection("SessionData");
+        var storeType = section.GetValue<string>("Using");
+
+        switch (storeType?.ToLower())
+        {
+            case "cosmos":
+                builder.Services.AddCosmosCache(opts =>
+                {
+                    var settings = section.GetSection("Settings").Get<CosmosCacheSettings>();
+                    ArgumentNullException.ThrowIfNull(settings);
+
+                    opts.ContainerName = settings.ContainerName ?? throw new ArgumentNullException(nameof(settings.ContainerName));
```

Comment:
Argument assertions could be moved to `ArgumentNullException.ThrowIfNull()` as per line 18

Reply:
Ah, another good spot

Reply:
Pushed changes